qiskit/transpiler/passes/group_gates.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Pass for grouping runs of two qubit gates and returning another graph with larger nodes.
+"""
+
+# This will be turned into an analysis pass.  At present this class is only used by a_star_cx.py
+
+
+def group_gates(compiled_dag):
+    """Group consecutive runs of gates on a qubit pair."""
+
+    import networkx as nx
+
+    gates = nx.topological_sort(compiled_dag.multi_graph)
```

Comment:
We are trying to avoid accessing `multi_graph` directly.  Consider using the DAG API or extend the DAG methods.